Source/qol/stash.h

```diff
@@ -68,30 +110,110 @@ constexpr Point InvalidStashPoint { -1, -1 };
+
 /**
  * @brief Render the inventory panel to the given buffer.
  */
 void DrawStash(const Surface &out);
+
+/**
+ * @brief Responds to action against a screen position within the stash panel (ex: mouse click).
+ * @param mousePosition x,y screen coordinates of mouse during action.
+ * @param isShiftHeld bool for whether shift key is down (auto transfer to belt).
+ * @param isCtrlHeld bool for whether ctrl key is down (auto transfer to inventory).
+ */
 void CheckStashItem(Point mousePosition, bool isShiftHeld = false, bool isCtrlHeld = false);
-bool UseStashItem(uint16_t cii);
+
+/**
+ * @brief Attempt to use item from stash.
+ * @param itemIndex 0-based index of the item as it exists in the stashList.
+ * @return 'True' if item used, 'False' if failure to use item.
+ */
+bool UseStashItem(uint16_t itemIndex);
+
+/**
+ * @brief Handle stash highlight and stats pop-up, if an item is under the cursor.
+ * @param mousePosition x,y screen coordinates of mouse.
+ * @return 0-based index of the item as it exists in the stashList, or InvalidItemId if nothing is under the cursor.
+ */
 uint16_t CheckStashHLight(Point mousePosition);
+
+/**
+ * @brief Perform UI action based on x,y screen position of mouse, responding to StashButtonPressed value.
+ * @param mousePosition x,y screen coordinates of mouse at release.
+ */
 void CheckStashButtonRelease(Point mousePosition);
+
+/**
+ * @brief Store x,y screen position of mouse, setting StashButtonPressed for use in CheckStashButtonRelease.
+ * @param mousePosition x,y screen coordinates of mouse at press.
+ */
 void CheckStashButtonPress(Point mousePosition);
 
+/**
+ * @brief Being the gold withdraw process, initializing the withdraw dialog popup.
+ */
 void StartGoldWithdraw();
+
+/**
+ * @brief Respond to key press while the gold withdraw process is in progress.
+ * @param vkey SDL keycode of the pressed key.
+ */
 void WithdrawGoldKeyPress(SDL_Keycode vkey);
+
+/**
+ * @brief Rendering the gold withdraw popup and the current value.
+ * @param out SDL surface upon which to draw the output.
+ * @param amount entered withdraw amount to populate in the withdraw popup.
+ */
 void DrawGoldWithdraw(const Surface &out, int amount);
+
+/**
+ * @brief Stop the withdraw process, ending text input, without gold transfer.
+ */
 void CloseGoldWithdraw();
+
+/**
+ * @brief Update gold withdraw amount in gold withdraw popup.
+ * @param text Text value of the entered withdraw amount.
+ */
 void GoldWithdrawNewText(string_view text);
 
+/**
+ * @brief Update the StashGrid on the provided page to account for item movement.
+ * @param page stashGrid page to act upon.
+ * @param startPoint x,y stashGrid position (from the bottom-left cell of the item!).
+ * @param itemSize x,y stashGrid size of the item (ex: 2x3).
+ * @param itemId item id exactly as it should be stored in the StashGrid.
+ */
+void UpdateStashGrid(unsigned page, Point startPoint, Size itemSize, StashStruct::StashCell itemId = 0);

```

Comment:
This is an internal function that operates on a `StashStruct` instance, it should be a private member of that class.

Reply:
Also should take `Rectangle` as a param instead of `Point, Size` separately.

Reply:
Moving it private is a fine suggestion. The reason for the separate point & size is the startPosition is move from the bottom-left cell to the top-left cell on [stash.cpp L640](https://github.com/hateborne/devilutionX/blob/nov2022_qol_tweak/Source/qol/stash.cpp#L640) (below). 
```cpp
void UpdateStashGrid(unsigned page, Point startPoint, Size itemSize, StashStruct::StashCell itemId /*= 0*/)
{
	if (itemId == InvalidItemId) {
		return;
	}

	// set start position to top-left corner as PointsInRectangle draws right (x) and down (y)
	Point adjustedStartPoint = startPoint - Displacement(0, itemSize.height - 1);

	for (Point point : PointsInRectangle(Rectangle { adjustedStartPoint, itemSize })) {
		Stash.stashGrids[page][point.x][point.y] = itemId;
	}
}
```

Reply:
Making this happen in next push (just a few min)